test/rubygems/test_gem.rb

```diff
@@ -1906,6 +1906,24 @@ def test_platform_defaults
     assert platform_defaults.is_a? Hash
   end
 
+  # Ensure that `Gem.source_date_epoch` is consistent even if
+  # $SOURCE_DATE_EPOCH has not been set.
+  def test_default_source_date_epoch_doesnt_change
+    old_epoch = ENV['SOURCE_DATE_EPOCH']
+    ENV['SOURCE_DATE_EPOCH'] = nil
+
+    # Unfortunately, there is no real way to test this aside from waiting
```

Comment:
There's `Time.now` being stubbed in `test/rubygems/test_gem_package_tar_writer.rb`. Any reason to not use this approach as well in here? Would be test too coupled with implementation?

Reply:
I suspect you may have commented on the wrong part of the code. Did you intend to comment on [`test/rubygems/test_gem.rb`](https://github.com/rubygems/rubygems/pull/3088/files#diff-6d1d09cbeb26123575a4efe6dc0ddab2R86)?

If you were referring to that, I did add that bit which resets `@default_source_date_epoch` to `test_gem_package_tar_writer.rb` for the same reason.

Reply:
I mean instead of using `sleep 1` stub `Time.now`.

Reply:
Ah, okay. I think that would be coupling it a bit too closely to the implementation, yeah.